sdk/core/azure-core-http-okhttp/src/test/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClientBuilderTests.java

```diff
@@ -195,6 +205,35 @@ public void buildWithConnectionTimeout() {
             .verifyComplete();
     }
 
+
+    /**
+     * Tests building a client with a given {@code connectionTimeout}.
+     */
+    @Test
+    public void buildWithFollowRedirect() {
```

Comment:
Mind splitting this into three tests?

`useDefaultFollowRedirect`
`buildWithFollowRedirect`
`buildWithoutFollowRedirect`

Right now were testing three different, but related, concepts in the same test which makes it harder to troubleshoot if the test begins failing as it could be a change in default behavior either on the builder or OkHttp HttpClient